### PR TITLE
Adjust field in LinkCapabilities struct

### PR DESCRIPTION
**Pull request by gferon**
This fixes linking.

### CODE REVIEW

File: libsignal-service/src/push_service.rs [hunk 1]
```diff
@@ -434,7 +434,14 @@ pub struct LinkAccountAttributes {
...
 #[serde(rename_all = "camelCase")]
 pub struct LinkCapabilities {
-    pub pni: bool,
+    pub delete_sync: bool,
+}
+
+// https://github.com/signalapp/Signal-Desktop/blob/1e57db6aa4786dcddc944349e4894333ac2ffc9e/ts/textsecure/WebAPI.ts#L1287
+impl Default for LinkCapabilities {
+    fn default() -> Self {
+        Self { delete_sync: true }
```
[the remark this answers is not in the document]
Comment (gferon) [reply libsignal-service/src/push_service.rs:443]: Fair, let me try if it works with `false` 😄

[the remark this answers is not in the document]
Comment (rubdos) [reply libsignal-service/src/push_service.rs:443]: Well, if we can just find what this means, and document it, that'd also be fine I suppose :)

[the remark this answers is not in the document]
Comment (direc85) [reply libsignal-service/src/push_service.rs:443]: I feel the `default()` is pointless here, we could just document the only place it's used in. And we need to know what the flag means before we set it either way.

[the remark this answers is not in the document]
Comment (gferon) [reply libsignal-service/src/push_service.rs:443]: I guess it's a matter of taste, I'm not attached to any way. I'll try to set the flag to `false` and ditch `Default`.

[the remark this answers is not in the document]
Comment (gferon) [reply libsignal-service/src/push_service.rs:443]: > Well, if we can just find what this means, and document it, that'd also be fine I suppose :)

Can confirm that you cannot link when setting the flag to `false`.

**Comment by direc85** [comment]
Just noticed this was merged already. Anyway...

I found out [what Signal Android does](https://github.com/signalapp/Signal-Android/blob/6e55bc04abe0b2c3029053d073fba9fc46a5fc76/app/src/main/java/org/thoughtcrime/securesms/jobs/MultiDeviceDeleteSyncJob.kt#L41):

```
ThreadDelete(
  threadRecipientId = threadRecipient.id.toLong(),
  isFullDelete = isFullDelete,
  messages = messages.map {
    AddressableMessage(
      sentTimestamp = it.dateSent,
      authorRecipientId = it.fromRecipient.id.toLong()
    )
  },
  nonExpiringMessages = nonExpiringMessages.map {
    AddressableMessage(
      sentTimestamp = it.dateSent,
      authorRecipientId = it.fromRecipient.id.toLong()
    )
  }
)
```

Whisperfish at least already supports individual message deletes, but this feature is about syncing thread deletes i.e. deleting full conversations across devices. That can't be done in libsignal IIUC and so has to be a flag provided by the client and not hardcoded in libsignal-service.

Please change the default to false and provide the API to set the flag. Thanks.

**Comment by gferon** [comment]
> Just noticed this was merged already. Anyway...
> 
> I found out [what Signal Android does](https://github.com/signalapp/Signal-Android/blob/6e55bc04abe0b2c3029053d073fba9fc46a5fc76/app/src/main/java/org/thoughtcrime/securesms/jobs/MultiDeviceDeleteSyncJob.kt#L41):
> 
> ```
> ThreadDelete(
>   threadRecipientId = threadRecipient.id.toLong(),
>   isFullDelete = isFullDelete,
>   messages = messages.map {
>     AddressableMessage(
>       sentTimestamp = it.dateSent,
>       authorRecipientId = it.fromRecipient.id.toLong()
>     )
>   },
>   nonExpiringMessages = nonExpiringMessages.map {
>     AddressableMessage(
>       sentTimestamp = it.dateSent,
>       authorRecipientId = it.fromRecipient.id.toLong()
>     )
>   }
> )
> ```
> 
> Whisperfish at least already supports individual message deletes, but this feature is about syncing thread deletes i.e. deleting full conversations across devices. That can't be done in libsignal IIUC and so has to be a flag provided by the client and not hardcoded in libsignal-service.
> 
> Please change the default to false and provide the API to set the flag. Thanks.

`false` is not an option, since you can't link without it: the support is now mandatory. I'll add a comment to explain the feature and remove `impl Default`.

**Pull request merged**